mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs

```diff
@@ -714,7 +714,8 @@ public override MethodInfo MakeGenericMethod (Type [] methodInstantiation)
 #if FULL_AOT_RUNTIME
 				throw new NotSupportedException ("User types are not supported under full aot");
 #else
-				return new MethodOnTypeBuilderInst (this, methodInstantiation);
+				if (RuntimeFeature.IsDynamicCodeSupported)
```

Comment:
I think it would be better to throw for `!RuntimeFeature.IsDynamicCodeSupported` and remove `#if FULL_AOT_RUNTIME`

```
throw new PlatfornNotSupportedException ("User types are not supported");
```